src/mon/NVMeofGwMon.cc

```diff
@@ -115,18 +115,25 @@ void NVMeofGwMon::tick()
       _propose_pending |= propose;
       last_beacon.erase(lb);
     } else {
-      BeaconSubsystems  *subsystems =
-         &pending_map.created_gws[lb.group_key][lb.gw_id].subsystems;
-      if (subsystems && subsystems->size() && old_group_key != lb.group_key) {
-        // to call track_deleting_gws once per each group-key
-        pending_map.track_deleting_gws(lb.group_key, *subsystems, propose);
-        old_group_key = lb.group_key;
-        _propose_pending |= propose;
-      }
       dout(20) << "beacon live for GW key: " << lb.gw_id << dendl;
     }
   }
-
+  BeaconSubsystems   empty_subsystems;
+  for (auto& group_state: pending_map.created_gws) {
+    auto& group_key = group_state.first;
+    auto& gws_states = group_state.second;
+    BeaconSubsystems  * subsystems = &empty_subsystems;
+    for (auto& gw_state : gws_states) { // loop for GWs inside nqn group
+      auto& gw_id = gw_state.first;
+      subsystems =
+        &pending_map.created_gws[group_key][gw_id].subsystems;
```

Comment:
Seems like you already have a reference to this -- gw_state.second.  It would probably make more sense to use a structured binding here as well

```for(auto &[gw_id, gw_state]...``` and simply use gw_state.subsystems here.

Reply:
agree